### PR TITLE
Fix: msan in UUIDStringToNum

### PR DESCRIPTION
**Pull request by devcrafter**
https://s3.amazonaws.com/clickhouse-test-reports/62004/eee4f9be1dc438ef5f73110c6c2a1306ef230fef/ast_fuzzer__msan_.html

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 669e3b44b0041db102c010bec2a69b8a1df269c3 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62610/669e3b44b0041db102c010bec2a69b8a1df269c3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Functions/FunctionsCodingUUID.cpp [hunk 1]
```diff
@@ -128,10 +128,14 @@ UUIDSerializer::Variant parseVariant(const DB::ColumnsWithTypeAndName & argument
     if (arguments.size() < 2)
         return UUIDSerializer::Variant::Default;
 
-    const auto representation = static_cast<magic_enum::underlying_type_t<UUIDSerializer::Variant>>(arguments[1].column->getInt(0));
-    const auto as_enum = magic_enum::enum_cast<UUIDSerializer::Variant>(representation);
+    Int64 value = static_cast<UInt64>(UUIDSerializer::Variant::Default);
+    if (!arguments[1].column->empty())
```
Comment (Algunenano) [inline src/Functions/FunctionsCodingUUID.cpp:132]: Maybe we should enforce that the column must be const. Looks like right now we are just taking the value of the first row (which caused the crash when empty), which will be confusing if somebody passes a full column as the variant argument

**Review by Algunenano — APPROVED** [review]

**Comment by devcrafter** [comment]
[AST Fuzzer (asan) failure](https://s3.amazonaws.com/clickhouse-test-reports/62610/a4a6581be354a5c5ed7953f5463ec655e9bcca4e/ast_fuzzer__asan_.html) - https://github.com/ClickHouse/ClickHouse/issues/56640

**Comment by rschu1ze** [comment]
Lol, I [fixed](https://github.com/ClickHouse/ClickHouse/pull/62852/commits/c820bc31e5e0b970384dfca858117d47b7de0f89) the same issue today without knowing about this PR.

**Pull request merged**